src/output/shell_output.go

```diff
@@ -686,13 +686,22 @@ func shouldInclude(file string, files []string) bool {
 		return true
 	}
 	for _, f := range files {
-		if file == f {
+		if file == f || wildcardMatch(file, f) {
```

Comment:
You should use `path.Match` for this (https://golang.org/pkg/path/#Match); that takes care of a few more patterns, and basically does the work for you :)

We also have internal code for double-star globs (in `src/fs/glob.go`) although I don't think it's factored in a way that's super convenient to use here so maybe leave for now.